WFLY-21041 JGroups subsystem documentation references deprecated/removed concepts

rhusar opened the pull request:
Resolves
https://issues.redhat.com/browse/WFLY-21041

What initially started as fix for legacy operations like :add-protocol turned out to be a complete rewrite since most sections were simply out of date. 

Note to reviewers: if reviewers want changes, please use **use code suggestions** or just submit PR against the topic branch.

- Remove legacy/non-existent operations and attributes
- Update attribute descriptions and add missing ones
- Add comprehensive missing sections for channels and channel configuration
- Add protocol removal examples and improve property configuration guidance
- Update CLI examples with complete stack creation workflows
- Restructure content with better formatting and line breaks
- Simplify verbose descriptions while maintaining technical accuracy

rhusar committed:
[WFLY-21041] Fix and improve JGroups subsystem documentation structure and clarity

- Remove legacy/non-existent operations and attributes
- Update attribute descriptions and add missing ones
- Add comprehensive missing sections for channels and channel configuration
- Add protocol removal examples and improve property configuration guidance
- Update CLI examples with complete stack creation workflows
- Restructure content with better formatting and line breaks
- Simplify verbose descriptions while maintaining technical accuracy

Signed-off-by: Radoslav Husar <radosoft@gmail.com>

bstansberry commented on docs/src/main/asciidoc/_high-availability/jgroups/JGroups_Subsystem.adoc:
```diff
@@ -12,83 +12,76 @@ endif::[]
-* allows definition of named protocol stacks
-* view run-time metrics associated with channels
-* specify a default stack for general use
+* allows definition of named protocol stacks and channels
```
This suggestion and the ones for the next two lines are about grammatical consistency among the 3 bullet points and with the preceding "the following features:".

bstansberry commented on docs/src/main/asciidoc/_high-availability/jgroups/JGroups_Subsystem.adoc:
```diff
@@ -97,161 +90,176 @@ directory.
-* `xmlns` This attribute specifies the XML namespace of the JGroups
-subsystem and, in particular, its version.
+* `default` is used to specify a default channel for the JGroups subsystem.
+This default channel will be used whenever a stack is required but no stack is specified.
```
I don't know if ^^^ is correct; it just seems more logical.

Also if it's possible to explain more about "is required" that would be good. Maybe "This default channel will be used whenever a channel is required by some service but no channel is specified in that service's configuration."

rhusar replied:
Yes, that's correct. I was rewriting an old sentence where we operated around stacks, rather than channels and did not change all the occurences. Thanks!

bstansberry commented on docs/src/main/asciidoc/_high-availability/jgroups/JGroups_Subsystem.adoc:
```diff
@@ -97,161 +90,176 @@ directory.
 === <stack>
 
-This element is used to configure a JGroups protocol stack.
+This element containing sequence of protocols is used to configure a JGroups protocol stack.
```
```suggestion
This element containing a sequence of protocols is used to configure a JGroups protocol stack.
```

bstansberry commented on docs/src/main/asciidoc/_high-availability/jgroups/JGroups_Subsystem.adoc:
```diff
@@ -97,161 +90,176 @@ directory.
-
-.
+Attributes site, rack and machine IDs are used by the Infinispan topology-aware consistent hash function,
+which when using dist mode, prevents dist mode replicas from being stored on the same host, rack or site.
```
```suggestion
which, when using dist mode, prevents dist mode replicas from being stored on the same host, rack or site.
```

bstansberry commented on docs/src/main/asciidoc/_high-availability/jgroups/JGroups_Subsystem.adoc:
```diff
@@ -97,161 +90,176 @@ directory.
+* `type` specifies the name of the JGroups protocol implementation (e.g. `MPING`, `pbcast.GMS`),
+with the package prefix `org.jgroups.protocols` removed.
+* `socket-binding` references a defined socket binding in the server profile.
+It is used when JGroups needs to create general sockets internally for this protocol instance.
```
The example uses the distinct 'socket-protocol' element for these.

bstansberry reviewed (changes requested):
Various copy edit suggestions.

Also, perhaps something like this at the end? 

https://github.com/wildfly/wildfly/blob/main/docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Jakarta_Batch.adoc?plain=1#L25-L26

Please don't block on my re-review of this if you get an approval from someone else. Of course I'm happy to look again if asked.

rhusar committed:
WFLY-21041 Incorporate remaining feedback from the the PR; add link to wildscribe and  documentation for socket, jdbc and socker-discovery types.

Update docs/src/main/asciidoc/_high-availability/jgroups/JGroups_Subsystem.adoc

Co-authored-by: Brian Stansberry <bstansberry@gmail.com>

Update docs/src/main/asciidoc/_high-availability/jgroups/JGroups_Subsystem.adoc

Co-authored-by: Brian Stansberry <bstansberry@gmail.com>

Update docs/src/main/asciidoc/_high-availability/jgroups/JGroups_Subsystem.adoc

Co-authored-by: Brian Stansberry <bstansberry@gmail.com>

Update docs/src/main/asciidoc/_high-availability/jgroups/JGroups_Subsystem.adoc

Co-authored-by: Brian Stansberry <bstansberry@gmail.com>

Update docs/src/main/asciidoc/_high-availability/jgroups/JGroups_Subsystem.adoc

Co-authored-by: Brian Stansberry <bstansberry@gmail.com>

Update docs/src/main/asciidoc/_high-availability/jgroups/JGroups_Subsystem.adoc

Co-authored-by: Brian Stansberry <bstansberry@gmail.com>

Update docs/src/main/asciidoc/_high-availability/jgroups/JGroups_Subsystem.adoc

Co-authored-by: Brian Stansberry <bstansberry@gmail.com>

Update docs/src/main/asciidoc/_high-availability/jgroups/JGroups_Subsystem.adoc

Co-authored-by: Brian Stansberry <bstansberry@gmail.com>

Update docs/src/main/asciidoc/_high-availability/jgroups/JGroups_Subsystem.adoc

Co-authored-by: Brian Stansberry <bstansberry@gmail.com>
Signed-off-by: Radoslav Husar <radosoft@gmail.com>

rhusar commented:
Thanks @bstansberry (really appreciate it!) and @pferraro !
